# Desafio Técnico - Cientista de Dados Júnior
### Localização de chamados do 1746

##### Autor: Brendow Paolillo Castro Isidoro
Fork: https://github.com/BrendowPaolillo-dev/emd-desafio-junior-data-scientist

---

Importando a biblioteca ``basedosdados``

In [2]:
import basedosdados as bd
import pandas as pd

Realizando a consulta e armazenando os dados de ``rede_hoteleira_ocupacao_eventos`` em um dataframe

Realizando a consulta e armazenando os dados de ``rede_hoteleira_ocupacao_eventos`` em um dataframe

In [3]:
query = "SELECT * FROM `datario.turismo_fluxo_visitantes.rede_hoteleira_ocupacao_eventos`"
df_hoteleira = bd.read_sql(query, billing_project_id="teste-dados-rio-brendow")

Downloading: 100%|██████████| 4/4 [00:00<00:00, 11.46rows/s]


Realizando a consulta e armazenando os dados de ``datario.administracao_servicos_publicos.chamado_1746`` em um dataframe

In [4]:
query = "SELECT * FROM `datario.administracao_servicos_publicos.chamado_1746` WHERE DATE(data_inicio) >= DATE('2022-01-01') AND DATE(data_inicio) < DATE('2023-01-01')"
df_chamado_sample_2022_2023 = bd.read_sql(query, billing_project_id="teste-dados-rio-brendow")

Downloading: 100%|██████████| 773524/773524 [05:12<00:00, 2473.30rows/s]


In [5]:
query = "SELECT * FROM `datario.administracao_servicos_publicos.chamado_1746` WHERE DATE(data_inicio) >= DATE('2023-01-01') AND DATE(data_inicio) <= DATE('2024-01-01')"
df_chamado_sample_2023_2024 = bd.read_sql(query, billing_project_id="teste-dados-rio-brendow")

Downloading: 100%|██████████| 844087/844087 [05:14<00:00, 2681.18rows/s]


Agrupando as fatias de dados do ``datario.administracao_servicos_publicos.chamado_1746`` em um dataframe

In [96]:
frames = [df_chamado_sample_2022_2023, df_chamado_sample_2023_2024]
 
df_chamado_sample = pd.concat(frames)
display(df_chamado_sample)

id_chamado         data_inicio            data_fim id_bairro  \
0        17341034 2022-06-06 22:38:56 2022-06-12 09:37:00         1   
1        17402357 2022-06-23 19:05:07 2022-06-30 23:41:00         2   
2        17372411 2022-06-15 12:39:22 2022-06-16 02:10:00         2   
3        17377922 2022-06-17 11:58:05 2022-06-17 13:20:00         3   
4        17406853 2022-06-24 21:07:06 2022-06-26 21:16:00         3   
...           ...                 ...                 ...       ...   
844082   18203640 2023-01-26 10:20:35 2023-01-26 18:31:57      None   
844083   18178283 2023-01-22 21:35:04 2023-01-26 18:15:31      None   
844084   18176000 2023-01-22 01:06:45 2023-01-26 18:12:45      None   
844085   18090257 2023-01-03 19:38:56 2023-01-09 13:47:54      None   
844086   18176211 2023-01-22 07:11:49 2023-01-26 18:20:43        61   

       id_territorialidade id_logradouro  numero_logradouro  \
0                        1         63867                120   
1                        1         66639                 50   
2                        1         63669                177   
3                        1         63305                 55   
4                        1         60079                 13   
...                    ...           ...                ...   
844082                None          None               <NA>   
844083                None          None               <NA>   
844084                None          None               <NA>   
844085                None          None               <NA>   
844086                   3         83691                 99   

       id_unidade_organizacional  \
0                              8   
1                              8   
2                              8   
3                              8   
4                              8   
...                          ...   
844082                      1713   
844083                      1713   
844084                      1713   
844085                      1713   
844086                      1713   

                              nome_unidade_organizacional  \
0       RIOLUZ - Companhia Municipal de Energia e Ilum...   
1       RIOLUZ - Companhia Municipal de Energia e Ilum...   
2       RIOLUZ - Companhia Municipal de Energia e Ilum...   
3       RIOLUZ - Companhia Municipal de Energia e Ilum...   
4       RIOLUZ - Companhia Municipal de Energia e Ilum...   
...                                                   ...   
844082  CEPIR - Coordenadoria Executiva de Promoção da...   
844083  CEPIR - Coordenadoria Executiva de Promoção da...   
844084  CEPIR - Coordenadoria Executiva de Promoção da...   
844085  CEPIR - Coordenadoria Executiva de Promoção da...   
844086  CEPIR - Coordenadoria Executiva de Promoção da...   

                            id_unidade_organizacional_mae  ... tempo_prazo  \
0       RIOLUZ - Companhia Municipal de Energia e Ilum...  ...        <NA>   
1       RIOLUZ - Companhia Municipal de Energia e Ilum...  ...        <NA>   
2       RIOLUZ - Companhia Municipal de Energia e Ilum...  ...        <NA>   
3       RIOLUZ - Companhia Municipal de Energia e Ilum...  ...        <NA>   
4       RIOLUZ - Companhia Municipal de Energia e Ilum...  ...        <NA>   
...                                                   ...  ...         ...   
844082  SEGOVI - Secretaria Municipal de Governo e Int...  ...        <NA>   
844083  SEGOVI - Secretaria Municipal de Governo e Int...  ...        <NA>   
844084  SEGOVI - Secretaria Municipal de Governo e Int...  ...        <NA>   
844085  SEGOVI - Secretaria Municipal de Governo e Int...  ...        <NA>   
844086  SEGOVI - Secretaria Municipal de Governo e Int...  ...        <NA>   

       prazo_unidade prazo_tipo   dentro_prazo   situacao tipo_situacao  \
0                  D          F  Fora do prazo  Encerrado      Atendido   
1                  D          F  Fora do prazo  Encerrado      Atendido   
2                  D          F       No prazo  Encerrado      Atendi

Ordenando dados por `"data_inicio"`

In [97]:
df_chamado_sample.sort_values(by=["data_inicio"])

id_chamado         data_inicio            data_fim id_bairro  \
423311   16725867 2022-01-01 00:00:15 2022-01-01 00:28:47       144   
439168   16725868 2022-01-01 00:05:50 2022-01-04 08:20:11       128   
442606   16725869 2022-01-01 00:13:17 2022-01-01 02:45:19        25   
396309   16725870 2022-01-01 00:16:14 2022-01-01 09:33:56      None   
383356   16725871 2022-01-01 00:19:41 2022-01-29 00:45:00       131   
...           ...                 ...                 ...       ...   
147222   19660604 2024-01-01 01:08:06                 NaT       141   
147221   19660605 2024-01-01 01:11:42                 NaT        43   
147227   19660607 2024-01-01 01:21:03 2024-01-01 01:21:03         8   
147218   19660609 2024-01-01 01:52:48                 NaT        37   
147216   19660610 2024-01-01 01:57:19                 NaT       157   

       id_territorialidade id_logradouro  numero_logradouro  \
423311                   5         12161                627   
439168                   4        204339               1014   
442606                   2         70854                 18   
396309                None          None                  1   
383356                   4         95497                 88   
...                    ...           ...                ...   
147222                   5         37762               5896   
147221                   3        121582                277   
147227                   1        211144                300   
147218                   2        108266                  1   
147216                   3        196121                  1   

       id_unidade_organizacional  \
423311                        70   
439168                        70   
442606                        70   
396309                        41   
383356                         8   
...                          ...   
147222                        70   
147221                        70   
147227                        41   
147218                        70   
147216                       118   

                              nome_unidade_organizacional  \
423311        GM-RIO - Guarda Municipal do Rio de Janeiro   
439168        GM-RIO - Guarda Municipal do Rio de Janeiro   
442606        GM-RIO - Guarda Municipal do Rio de Janeiro   
396309  DEFESA CIVIL - Subsecretaria de Proteção e Def...   
383356  RIOLUZ - Companhia Municipal de Energia e Ilum...   
...                                                   ...   
147222        GM-RIO - Guarda Municipal do Rio de Janeiro   
147221        GM-RIO - Guarda Municipal do Rio de Janeiro   
147227  DEFESA CIVIL - Subsecretaria de Proteção e Def...   
147218        GM-RIO - Guarda Municipal do Rio de Janeiro   
147216    COMLURB - Companhia Municipal de Limpeza Urbana   

                            id_unidade_organizacional_mae  ... tempo_prazo  \
423311        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
439168        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
442606        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
396309  DEFESA CIVIL - Subsecretaria de Proteção e Def...  ...        <NA>   
383356  RIOLUZ - Companhia Municipal de Energia e Ilum...  ...        <NA>   
...                                                   ...  ...         ...   
147222        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
147221        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
147227  DEFESA CIVIL - Subsecretaria de Proteção e Def...  ...        <NA>   
147218        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
147216    COMLURB - Companhia Municipal de Limpeza Urbana  ...        <NA>   

       prazo_unidade prazo_tipo   dentro_prazo       situacao  \
423311             H          F       No prazo      Encerrado   
439168             H          F  Fora do prazo      Encerrado   
442606             H          F       No prazo      Encerrado   
396309             D          F   

---
### Questão 1

Quantos chamados foram abertos no dia 01/04/2023?

Realizando a consulta e armazenando os dados de ``chamado_1746`` em um dataframe, estes que são referentes ao dia ``01/04/2023``:

In [98]:
df_chamado_01042023=df_chamado_sample.query('"2023-04-01 00:00:00" <= data_inicio < "2023-04-02 00:00:00"').sort_values(by=["data_inicio"])
df_chamado_01042023

id_chamado         data_inicio data_fim id_bairro id_territorialidade  \
439465   18516181 2023-04-01 00:00:37      NaT        26                   2   
480102   18516182 2023-04-01 00:00:55      NaT       144                   5   
471114   18516183 2023-04-01 00:02:05      NaT       127                   4   
455782   18516185 2023-04-01 00:03:13      NaT        80                   3   
428894   18516186 2023-04-01 00:03:51      NaT       161                   1   
...           ...                 ...      ...       ...                 ...   
461598   18516259 2023-04-01 01:19:36      NaT        55                   3   
441072   18516261 2023-04-01 01:23:20      NaT        26                   2   
462970   18516264 2023-04-01 01:26:17      NaT        55                   3   
462971   18516265 2023-04-01 01:26:35      NaT        55                   3   
481570   18516268 2023-04-01 01:40:40      NaT       144                   5   

       id_logradouro  numero_logradouro id_unidade_organizacional  \
439465         69591                259                        70   
480102         27730                132                         8   
471114        162859                  0                       922   
455782         43554                 54                       178   
428894         61325                200                        70   
...              ...                ...                       ...   
461598         59899                 87                       172   
441072         69591                541                        70   
462970         59899                  2                       776   
462971         59899                  2                       776   
481570         17152               3006                        82   

                              nome_unidade_organizacional  \
439465        GM-RIO - Guarda Municipal do Rio de Janeiro   
480102  RIOLUZ - Companhia Municipal de Energia e Ilum...   
471114                             Central de Ambulâncias   
455782                                              13aGC   
428894        GM-RIO - Guarda Municipal do Rio de Janeiro   
...                                                   ...   
461598                                              07aGC   
441072        GM-RIO - Guarda Municipal do Rio de Janeiro   
462970                                        GLF – Meier   
462971                                        GLF – Meier   
481570  CLF - Coordenadoria de Licenciamento e Fiscali...   

                            id_unidade_organizacional_mae  ... tempo_prazo  \
439465        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
480102  RIOLUZ - Companhia Municipal de Energia e Ilum...  ...        <NA>   
471114                SMS - Secretaria Municipal de Saúde  ...        <NA>   
455782                                         SECONSERVA  ...        <NA>   
428894        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
...                                                   ...  ...         ...   
461598                                         SECONSERVA  ...        <NA>   
441072        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
462970  SMDEIS - Secretaria Municipal de Desenvolvimen...  ...        <NA>   
462971  SMDEIS - Secretaria Municipal de Desenvolvimen...  ...        <NA>   
481570  CLF - Coordenadoria de Licenciamento e Fiscali...  ...        <NA>   

       prazo_unidade prazo_tipo   dentro_prazo       situacao tipo_situacao  \
439465             H          F  Fora do prazo  Não Encerrado     Andamento   
480102             D          F  Fora do prazo  Não Encerrado     Andamento   
471114             H          F  Fora do prazo  Não Encerrado     Andamento   
455782             D          F  Fora do prazo  Não Encerrado     Andamento   
428894             D          F  Fora do prazo  Não Encerrado     Andamento   
...              ...        ...            ...            ...           .

Realizando a localização dos dados utilizando outro método:

In [99]:
df_chamado_sample.loc[(df_chamado_sample['data_inicio'] >= pd.to_datetime('2023-04-01')) & (df_chamado_sample['data_inicio'] < pd.to_datetime('2023-04-02'))].sort_values(by=["data_inicio"])

id_chamado         data_inicio data_fim id_bairro id_territorialidade  \
439465   18516181 2023-04-01 00:00:37      NaT        26                   2   
480102   18516182 2023-04-01 00:00:55      NaT       144                   5   
471114   18516183 2023-04-01 00:02:05      NaT       127                   4   
455782   18516185 2023-04-01 00:03:13      NaT        80                   3   
428894   18516186 2023-04-01 00:03:51      NaT       161                   1   
...           ...                 ...      ...       ...                 ...   
461598   18516259 2023-04-01 01:19:36      NaT        55                   3   
441072   18516261 2023-04-01 01:23:20      NaT        26                   2   
462970   18516264 2023-04-01 01:26:17      NaT        55                   3   
462971   18516265 2023-04-01 01:26:35      NaT        55                   3   
481570   18516268 2023-04-01 01:40:40      NaT       144                   5   

       id_logradouro  numero_logradouro id_unidade_organizacional  \
439465         69591                259                        70   
480102         27730                132                         8   
471114        162859                  0                       922   
455782         43554                 54                       178   
428894         61325                200                        70   
...              ...                ...                       ...   
461598         59899                 87                       172   
441072         69591                541                        70   
462970         59899                  2                       776   
462971         59899                  2                       776   
481570         17152               3006                        82   

                              nome_unidade_organizacional  \
439465        GM-RIO - Guarda Municipal do Rio de Janeiro   
480102  RIOLUZ - Companhia Municipal de Energia e Ilum...   
471114                             Central de Ambulâncias   
455782                                              13aGC   
428894        GM-RIO - Guarda Municipal do Rio de Janeiro   
...                                                   ...   
461598                                              07aGC   
441072        GM-RIO - Guarda Municipal do Rio de Janeiro   
462970                                        GLF – Meier   
462971                                        GLF – Meier   
481570  CLF - Coordenadoria de Licenciamento e Fiscali...   

                            id_unidade_organizacional_mae  ... tempo_prazo  \
439465        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
480102  RIOLUZ - Companhia Municipal de Energia e Ilum...  ...        <NA>   
471114                SMS - Secretaria Municipal de Saúde  ...        <NA>   
455782                                         SECONSERVA  ...        <NA>   
428894        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
...                                                   ...  ...         ...   
461598                                         SECONSERVA  ...        <NA>   
441072        GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
462970  SMDEIS - Secretaria Municipal de Desenvolvimen...  ...        <NA>   
462971  SMDEIS - Secretaria Municipal de Desenvolvimen...  ...        <NA>   
481570  CLF - Coordenadoria de Licenciamento e Fiscali...  ...        <NA>   

       prazo_unidade prazo_tipo   dentro_prazo       situacao tipo_situacao  \
439465             H          F  Fora do prazo  Não Encerrado     Andamento   
480102             D          F  Fora do prazo  Não Encerrado     Andamento   
471114             H          F  Fora do prazo  Não Encerrado     Andamento   
455782             D          F  Fora do prazo  Não Encerrado     Andamento   
428894             D          F  Fora do prazo  Não Encerrado     Andamento   
...              ...        ...            ...            ...           .

``Resposta:`` Como cada linha da consulta é referente a um chamado realizado no dia, logo, podemos verificar que houveram 73 chamados no dia 01/04/2023

---
### Questão 2

Qual o tipo de chamado que teve mais reclamações no dia 01/04/2023?

Primeiro, vejamos quais são as colunas existentes no nosso dataframe:

In [100]:
sorted(df_chamado_sample.keys())

['categoria',
 'data_alvo_diagnostico',
 'data_alvo_finalizacao',
 'data_fim',
 'data_inicio',
 'data_particao',
 'data_real_diagnostico',
 'dentro_prazo',
 'geometry',
 'id_bairro',
 'id_chamado',
 'id_logradouro',
 'id_subtipo',
 'id_territorialidade',
 'id_tipo',
 'id_unidade_organizacional',
 'id_unidade_organizacional_mae',
 'justificativa_status',
 'latitude',
 'longitude',
 'nome_unidade_organizacional',
 'numero_logradouro',
 'prazo_tipo',
 'prazo_unidade',
 'reclamacoes',
 'situacao',
 'status',
 'subtipo',
 'tempo_prazo',
 'tipo',
 'tipo_situacao',
 'unidade_organizacional_ouvidoria']

De acordo com a descrição no GCP, podemos concluir que a coluna ``'tipo'`` conterá a informação para responder esta questão.

In [101]:
df_chamado_01042023['tipo'].value_counts()

Poluição sonora             24
Estacionamento Irregular     9
Iluminação Pública           8
Remoção Gratuita             5
Limpeza                      5
Postura Municipal            4
Ocupação de área pública     3
Pavimentação                 3
Comlurb - Vetores            2
Fiscalização de obras        2
Manejo Arbóreo               2
Feiras                       1
Semáforo                     1
Ônibus                       1
Programa Cegonha Carioca     1
Drenagem e Saneamento        1
Comércio ambulante           1
Name: tipo, dtype: int64

``Resposta:`` Realizando a contagem dos valores repetidos na coluna ``'tipo'`` do nosso dataframe, podemos verificar que o tipo de chamado mais frequente no dia 01/04/2023 é ``Poluição sonora``, contendo 24 chamados

---
### Questão 3

Quais os nomes dos 3 bairros que mais tiveram chamados abertos nesse dia?

Realizando a consulta e armazenando os dados de ``bairro`` em um dataframe:

In [102]:
query = "SELECT * FROM `datario.dados_mestres.bairro`"
df_bairro = bd.read_sql(query, billing_project_id="teste-dados-rio-brendow")

Downloading: 100%|██████████| 164/164 [00:11<00:00, 14.12rows/s]


Verificando os ``'id_bairro'`` que mais se repetem entre os chamados:

In [103]:
id_bairros_max_chamados = df_chamado_01042023['id_bairro'].value_counts().nlargest(3).index.tolist()
id_bairros_max_chamados

['66', '26', '144']

In [104]:
bairros_max_chamados = df_bairro.loc[df_bairro['id_bairro'].isin(id_bairros_max_chamados)]['nome'].tolist()
bairros_max_chamados

['Leblon', 'Engenho de Dentro', 'Campo Grande']

``Resposta:`` Os bairros que mais possuem chamados no dia 01/04/2023 são: ``'Leblon'``, ``'Engenho de Dentro'``, ``'Campo Grande'``

---
### Questão 4

Qual o nome da subprefeitura com mais chamados abertos nesse dia?

In [105]:
subprefeitura_max_chamados = df_bairro.loc[df_bairro['id_bairro'].isin([id_bairros_max_chamados[0]])]['subprefeitura'].tolist()
subprefeitura_max_chamados

['Zona Norte']

``Resposta:`` A subprefeitura com maior número de chamados no dia 01/04/2023 foi a da ``'Zona Norte'``

---
### Questão 5

Existe algum chamado aberto nesse dia que não foi associado a um bairro ou subprefeitura na tabela de bairros? Se sim, por que isso acontece?

In [106]:
chamado_no_b_or_sub = df_chamado_01042023[df_chamado_01042023['id_bairro'].isna()].to_dict()
{'tipo': list(chamado_no_b_or_sub['tipo'].values())[0],'subtipo': list(chamado_no_b_or_sub['subtipo'].values())[0]}

{'tipo': 'Ônibus',
 'subtipo': 'Verificação de ar condicionado inoperante no ônibus'}

``Resposta:`` O chamado não foi associado a um bairro ou subprefeitura pois trata-se de um registro referente à ``'Verificação de ar condicionado inoperante no ônibus'``, logo não podendo ser atribuído a nenhuma localidade.

---
### Questão 6

Quantos chamados com o subtipo "Perturbação do sossego" foram abertos desde 01/01/2022 até 31/12/2023 (incluindo extremidades)?

In [107]:
df_chamado_sample.query('subtipo == "Perturbação do sossego" & "2022-01-01 00:00:00" <= data_inicio < "2024-01-01 00:00:00"').sort_values(by=["data_inicio"])

id_chamado         data_inicio            data_fim id_bairro  \
437651   16725889 2022-01-01 01:43:06 2022-01-17 09:43:34        20   
437853   16725896 2022-01-01 02:00:30 2022-01-17 09:44:38        26   
444545   16725898 2022-01-01 02:07:16 2022-01-01 19:04:28       128   
437980   16725902 2022-01-01 02:21:06 2022-01-04 08:31:58        38   
437846   16725903 2022-01-01 02:24:29 2022-01-04 14:21:03        26   
...           ...                 ...                 ...       ...   
146823   19572329 2023-12-08 20:09:40                 NaT        36   
146825   19579903 2023-12-11 13:45:30                 NaT        24   
146822   19580880 2023-12-11 15:51:25                 NaT        14   
146828   19599401 2023-12-14 16:35:34                 NaT        26   
146827   19613996 2023-12-18 16:48:29                 NaT        36   

       id_territorialidade id_logradouro  numero_logradouro  \
437651                   2         77172                 18   
437853                   2         77099                419   
444545                   4        134171                751   
437980                   2         10835                194   
437846                   2          3368                326   
...                    ...           ...                ...   
146823                   2         81737                 18   
146825                   2         64428                 88   
146822                   1         69575                 39   
146828                   2         66647               <NA>   
146827                   2         82149                364   

       id_unidade_organizacional                  nome_unidade_organizacional  \
437651                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
437853                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
444545                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
437980                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
437846                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
...                          ...                                          ...   
146823                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
146825                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
146822                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
146828                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
146827                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   

                      id_unidade_organizacional_mae  ... tempo_prazo  \
437651  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
437853  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
444545  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
437980  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
437846  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
...                                             ...  ...         ...   
146823  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
146825  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
146822  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
146828  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
146827  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   

       prazo_unidade prazo_tipo   dentro_prazo       situacao   tipo_situacao  \
437651             D          F  Fora do prazo      Encerrado    Não atendido   
437853             D          F  Fora do prazo      Encerrado    Não atendido   
444545             D          F       No prazo      Encerrado  Não constatado   
437980             D          F       No prazo      Encerrado    Não atendido   
437846             D          F       No prazo      Encerrado    Não atendido   
...              ...        ...            ...            ...            

In [108]:
df_chamado_sample.loc[(df_chamado_sample['subtipo'] == "Perturbação do sossego") & (df_chamado_sample['data_inicio'] >= "2022-01-01 00:00:00") & (df_chamado_sample['data_inicio'] < "2024-01-01 00:00:00")].sort_values(by=["data_inicio"])

id_chamado         data_inicio            data_fim id_bairro  \
437651   16725889 2022-01-01 01:43:06 2022-01-17 09:43:34        20   
437853   16725896 2022-01-01 02:00:30 2022-01-17 09:44:38        26   
444545   16725898 2022-01-01 02:07:16 2022-01-01 19:04:28       128   
437980   16725902 2022-01-01 02:21:06 2022-01-04 08:31:58        38   
437846   16725903 2022-01-01 02:24:29 2022-01-04 14:21:03        26   
...           ...                 ...                 ...       ...   
146823   19572329 2023-12-08 20:09:40                 NaT        36   
146825   19579903 2023-12-11 13:45:30                 NaT        24   
146822   19580880 2023-12-11 15:51:25                 NaT        14   
146828   19599401 2023-12-14 16:35:34                 NaT        26   
146827   19613996 2023-12-18 16:48:29                 NaT        36   

       id_territorialidade id_logradouro  numero_logradouro  \
437651                   2         77172                 18   
437853                   2         77099                419   
444545                   4        134171                751   
437980                   2         10835                194   
437846                   2          3368                326   
...                    ...           ...                ...   
146823                   2         81737                 18   
146825                   2         64428                 88   
146822                   1         69575                 39   
146828                   2         66647               <NA>   
146827                   2         82149                364   

       id_unidade_organizacional                  nome_unidade_organizacional  \
437651                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
437853                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
444545                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
437980                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
437846                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
...                          ...                                          ...   
146823                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
146825                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
146822                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
146828                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
146827                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   

                      id_unidade_organizacional_mae  ... tempo_prazo  \
437651  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
437853  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
444545  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
437980  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
437846  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
...                                             ...  ...         ...   
146823  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
146825  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
146822  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
146828  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   
146827  GM-RIO - Guarda Municipal do Rio de Janeiro  ...        <NA>   

       prazo_unidade prazo_tipo   dentro_prazo       situacao   tipo_situacao  \
437651             D          F  Fora do prazo      Encerrado    Não atendido   
437853             D          F  Fora do prazo      Encerrado    Não atendido   
444545             D          F       No prazo      Encerrado  Não constatado   
437980             D          F       No prazo      Encerrado    Não atendido   
437846             D          F       No prazo      Encerrado    Não atendido   
...              ...        ...            ...            ...            

``Resposta:`` O resultado da consulta contém 42408 chamados com o subtipo "Perturbação do sossego".

---
### Questão 7

Selecione os chamados com esse subtipo que foram abertos durante os eventos contidos na tabela de eventos (Reveillon, Carnaval e Rock in Rio).

In [109]:
df_chamado_perturbacao = df_chamado_sample.query('subtipo == "Perturbação do sossego" & "2022-01-01 00:00:00" <= data_inicio < "2024-01-01 00:00:00"').sort_values(by=["data_inicio"])

df_merge = df_chamado_perturbacao.reset_index().assign(key=1).merge(df_hoteleira.assign(key=1), on='key').drop('key', axis=1) 
df_chamado_eventos = df_merge[df_merge['data_inicio'].between(pd.to_datetime(df_merge['data_inicial']), pd.to_datetime(df_merge['data_final'])+ pd.Timedelta(days=1))]
df_chamado_eventos

index id_chamado         data_inicio            data_fim id_bairro  \
85636   228986   17658238 2022-09-02 00:00:50 2022-09-09 07:45:50       141   
85640   228836   17658242 2022-09-02 00:11:44 2022-09-02 04:08:22       132   
85644   227880   17658243 2022-09-02 00:12:20 2022-09-09 07:47:07        35   
85648   229110   17658249 2022-09-02 00:26:20 2022-09-09 07:48:09       161   
85652   229106   17658250 2022-09-02 00:26:41 2022-09-09 07:48:53       161   
...        ...        ...                 ...                 ...       ...   
138407  182674   18334673 2023-02-21 22:19:48 2023-02-21 23:08:46        40   
138411  229818   18334681 2023-02-21 22:31:44 2023-02-25 22:00:40        16   
138415  182851   18334715 2023-02-21 23:23:00 2023-02-22 04:07:18        40   
138419  229805   18334721 2023-02-21 23:33:14 2023-02-25 22:04:32        15   
138423  229823   18334727 2023-02-21 23:44:18 2023-02-25 21:52:18        33   

       id_territorialidade id_logradouro  numero_logradouro  \
85636                    5         36590                485   
85640                    4        215699                313   
85644                    2         81141                342   
85648                    1         61630                  0   
85652                    1         61630                  0   
...                    ...           ...                ...   
138407                   3         67025                355   
138411                   2         80960                102   
138415                   3         82438                 10   
138419                   2         76661                182   
138423                   2         73353                323   

       id_unidade_organizacional                  nome_unidade_organizacional  \
85636                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
85640                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
85644                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
85648                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
85652                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
...                          ...                                          ...   
138407                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
138411                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
138415                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
138419                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
138423                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   

        ...   tipo_situacao justificativa_status reclamacoes data_particao  \
85636   ...    Não atendido                 None           0    2022-09-01   
85640   ...    Não atendido                 None           0    2022-09-01   
85644   ...    Não atendido                 None           0    2022-09-01   
85648   ...    Não atendido                 None           0    2022-09-01   
85652   ...    Não atendido                 None           0    2022-09-01   
...     ...             ...                  ...         ...           ...   
138407  ...  Não constatado                 None           0    2023-02-01   
138411  ...    Não atendido                 None           0    2023-02-01   
138415  ...  Não constatado                 None           0    2023-02-01   
138419  ...    Não atendido                 None           0    2023-02-01   
138423  ...    Não atendido                 None           0    2023-02-01   

                              geometry                    ano data_inicial  \
85636   POINT(-43.4673957 -22.8847528)  02/09 a 04/09 de 2022   2022-09-02   
85640   POINT(-43.4751068 -23.0124709)  02/09 a 04/09 de 2022   2022-09-02   
85644   POINT(-43.2339442 -22.9222289)  02/09 a 04/09 de 2022   2022-09-02   
85648    POINT(-43.1861477 -22.912313)  02/09 a 04/09 de 2022   2022-09-02   
856

``Resposta:`` Foram encontrados ``1212`` registros com o subtipo "Perturbação do sossego" durante os eventos.

---
### Questão 8

Quantos chamados desse subtipo foram abertos em cada evento?

In [110]:
display(df_chamado_eventos.groupby(['evento'])['evento'].count().sort_values(ascending=False))

evento
Rock in Rio    834
Carnaval       241
Reveillon      137
Name: evento, dtype: int64

``Resposta:`` Foram encontrados 1212 registros com o subtipo "Perturbação do sossego" durante os eventos, sendo ``241`` no Carnaval, ``137`` no Reveillon, ``834`` no Rock in Rio.

---
### Questão 9

Qual evento teve a maior média diária de chamados abertos desse subtipo?

In [111]:
df_chamado_eventos.groupby(['evento', pd.Grouper(key='data_inicio', freq='D')]).size().groupby('evento').mean()

evento
Carnaval        60.250000
Reveillon       45.666667
Rock in Rio    119.142857
dtype: float64

``Resposta:`` A maior média diária é de ``119.14`` chamados com o subtipo "Perturbação do sossego" durante o Rock in Rio.

---
### Questão 10

Compare as médias diárias de chamados abertos desse subtipo durante os eventos específicos (Reveillon, Carnaval e Rock in Rio) e a média diária de chamados abertos desse subtipo considerando todo o período de 01/01/2022 até 31/12/2023.

In [112]:
df_chamado_perturbacao.shape[0] / ((pd.to_datetime('2024-01-01') - pd.to_datetime('2022-01-01')).days)

58.09315068493151

``Resposta:`` Os eventos possuem uma média diária de chamados de:<br/>
<br/>
       Evento    | Média <br/>
    ------------------------<br/>
      Rock in Rio | 119.5<br/>
      Carnaval    | 60.25<br/>
      Reveillon   | 45.6<br/>
<br/>
  Já a média diária de chamados durante todo o período de 01/01/2022 até 31/12/2023 é de: 58.09